test/test_nestedtensor.py

```diff
@@ -8435,15 +8401,16 @@ def test_forward(self, device, dtype, op):
     @ops(
         [op for op in njt_op_db if op.supports_njt and op.supports_autograd],
         allowed_dtypes=(torch.float32,),
+        sample_skips_and_xfails=BACKWARD_SKIPS_AND_XFAILS,
```

Comment:
General feedback: I can see this being useful. It's not directly useful right now my usecases (un-skipping some functorch tests). That being said it's probably not hard to build this in to that.

Some questions I have are: skips and xfails are generally on the OpInfo objects themselves. Are you thinking of having a variant that can look like that? It would be cool to have a single decorator you can apply to a Function for an OpInfo that makes one of these rules

Reply:
I updated the PR to store skips / xfails on the test function via the `@sample_skips_and_xfails` decorator. `@ops` reads and filters these, placing skips / xfails relevant to each op onto the parametrized test functions. Then the sample input generation funcs (e.g. `op.sample_inputs()`) pull these off of the test function to establish the right subtest context.

There are minimal changes to the test logic needed now, but still some: it's necessary to pass `op.sample_inputs(..., use_subtests=True)` when skips / xfails are defined so they can be handled appropriately. Then the iterator returns `(sample, subtest_ctx)` and the subtest_ctx must be entered.

> Some questions I have are: skips and xfails are generally on the OpInfo objects themselves. Are you thinking of having a variant that can look like that? It would be cool to have a single decorator you can apply to a Function for an OpInfo that makes one of these rules

With the above changes, it's possible to define an `xfailIf` decorator that is just syntactic sugar over the rule system. I added an example using this for un-skipping a vmap test with a more granular xfail based on a `SampleInput` property.